assignment2/BinaryTree.java

```diff
@@ -0,0 +1,198 @@
+package com.katermar.assignment2;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Created by katermar on 5/23/2017.
+ */
+public class BinaryTree {
+    TreeNode root;
+
+    /*Inner class for Binary Tree node
+    * @field value stores integer value
+    * @field left, right store left and right subtrees*/
+    public static class TreeNode {
+        private final int value;
+        private TreeNode left;
+        private TreeNode right;
+
+        TreeNode(int item) {
+            value = item;
+        }
+
+        public int getValue() {
+            return value;
+        }
+
+        public TreeNode getLeft() {
+            return left;
+        }
+
+        @Override
+        public String toString() {
+            return String.format("TreeNode = %d", this.value);
+        }
+
+        public TreeNode getRight() {
+            return right;
+        }
+
+    }
+
+    /* Function to print all ancestors of purposed value
+    * @param node - root node
+    * @param value - value of the node on which we want to print ancestors
+    * @return true, if value exists in this tree, otherwise - false*/
+    boolean findAllAncestors(List<TreeNode> list, TreeNode node, int value)
+    {
+         //if tree is empty returns false
+        if (node == null) {
+            return false;
+        }
+
+        //if value exist in a tree
+        if (node.getValue() == value) {
+            return true;
+        }
+
+        //if value is present in left/right subtree
+        // then we'll print it
+        if (findAllAncestors(list, node.left, value) || findAllAncestors(list, node.right, value)) {
+            list.add(node);
+            return true;
+        }
+
+        //else return false
+        return false;
+    }
+
+    /*Traverse tree in preorder
+    * @param root - tree root*/
+    public void preorderTraversal (TreeNode root){ //метод для обхода в ширину

```

Comment:
Minor comment regarding design: preorderTraversal is a method of BinaryTree and I would expect it to operate on the whole tree. As such the root parameter is not needed and makes method more difficult to use.

Better design would be BinaryRoot.preorderTraversal() without parameters calling TreeNode.preorderTraversal() (also without parameters - in such case root == this), which then descends recursively.